identity/src/secp256k1.rs

```diff
@@ -131,13 +131,7 @@ impl SecretKey {
     ///
     /// [RFC3278]: https://tools.ietf.org/html/rfc3278#section-8.2
     pub fn sign(&self, msg: &[u8]) -> Vec<u8> {
-        let generic_array = Sha256::digest(msg);
-
-        // FIXME: Once `generic-array` hits 1.0, we should be able to just use `Into` here.
-        let mut array = [0u8; 32];
-        array.copy_from_slice(generic_array.as_slice());
-
-        let message = Message::parse(&array);
+        let message = Message::parse(&Sha256::digest(msg).into());
```

Comment:
I wonder if we should have it in separate variables before passing it to `Message::parse`

eg

```suggestion
        let digest = Sha256::digest(msg);
        let message = Message::parse(&digest.into());
```

Thoughts? Not that it should impact anything from a glance over, but just thinking about readability.

Reply:
I believe the information is sufficient since the method is called `digest`. Calling `into` introduces some ambiguity but I don't think it matters.